frontend/client_sdk/src/CodaSDK.d.ts

```diff
@@ -0,0 +1,78 @@
+export declare type publicKey = string;
+export declare type privateKey = string;
+export declare type globalSlot = number;
+export declare type uint64 = string;
+export declare type key = {
+    readonly privateKey: privateKey;
+    readonly publicKey: publicKey;
+};
+export declare type signature = {
+    readonly field: string;
+    readonly scalar: string;
+};
+export declare type signed<signable> = {
+    readonly publicKey: publicKey;
+    readonly signature: signature;
+    readonly payload: signable;
+};
+export declare type stakeDelegation = {
+    readonly to: publicKey;
+    readonly from: publicKey;
+    readonly fee: uint64;
+    readonly nonce: number;
+    readonly memo?: string;
+    readonly validUntil: globalSlot;
+};
+export declare type payment = {
+    readonly to: publicKey;
+    readonly from: publicKey;
+    readonly fee: uint64;
+    readonly amount: uint64;
+    readonly nonce: number;
+    readonly memo?: string;
+    readonly validUntil: globalSlot;
+};
+/**
+  * Generates a public/private keypair
+  */
+export declare const genKeys: () => key;
+/**
+  * Signs an arbitrary message
+  *
+  * @param key - The keypair used to sign the message
+  * @param message - An arbitrary string message to be signed
+  * @returns A signed message
+  */
+export declare const signMessage: (message: string, key: key) => signed<string>;
+/**
+  * Verifies that a signature matches a message.
+  *
+  * @param signedMessage - A signed message
+  * @returns True if the `signedMessage` contains a valid signature matching
+  * the message and publicKey.
+  */
+export declare const verifyMessage: (signedMessage: signed<string>) => boolean;
```

Comment:
2 questions:
1) would `verifySignature` be more semantically correct?
2) why don't we need to pass the message and the pk into this function?

Reply:
I think I understand #2 -- because the input is a signed<x> which contains the signature, message and pk

This leads me to wonder if this may be a weird interface for most people, since you would have to manually construct the signed object before invoking this if you just received a signature from somebody. Am I thinking about this incorrectly?

Reply:
I saw that you have `signString`, maybe we just make it as simple as passing codaSDK into that  and call it a day? wdyt

eg `let signMessage = (. message: string, privateKey: string) => signString(codaSDK, privateKey, message)`

Reply:
All of the fields in signed<T> are necessary for checking that the signature is correct, passing an object is not an uncommon way of simulating named args in JS so I don't think it's so bad even in that case:

```js
verifyMessage({
  payload: "hello",
  publicKey: myPublicKey,
  signature
})
```

Reply:
yea, makes sense -- but two counterpoints:
- sometimes you may not want to send them the payload and signature together, in which case, you'd have to grab the sig from the object
- the object property names might be confusing, but I think it makes sense. Either way, you'd have to know which order to pass them into the function anyway, unless we used named args. The only one I have doubt about is payload -- as I'm used to thinking `verify(msg, sig, pk)`

Either way, this is fine -- I'm curious what other similar interfaces do